eng/scripts/Language-Settings.ps1

```diff
@@ -355,59 +355,99 @@ function PackageDependenciesResolve($artifactNamePrefix, $packageDirectory) {
-    -and (PackageDependenciesResolve $artifactNamePrefix $workingDirectory)
+function FallbackValidation ($packageInfos, $workingDirectory) {
+  $results = @()
+
+  foreach ($packageInfo in $packageInfos) {
+    $groupId = $packageInfo.Group
+    $artifactId = $packageInfo.Name
+    $version = $packageInfo.Version
+
+    Write-Host "Validating using mvn command directly on $artifactId."
+
+    $artifactNamePrefix = "${groupId}:${artifactId}:${version}"
+
+    $packageDirectory = Join-Path $workingDirectory "${groupId}__${artifactId}__${version}"
+    New-Item -ItemType Directory -Path $packageDirectory -Force | Out-Null
+
+    $isValid = (SourcePackageHasComFolder $artifactNamePrefix $packageDirectory) `
+      -and (PackageDependenciesResolve $artifactNamePrefix $packageDirectory)
+      
+    if (!$isValid) {
+      LogWarning "Package $artifactNamePrefix ref docs validation failed."
+    }
+
+    $results += $isValid
+  }
+
+  $allValid = $results.Where({ $_ -eq $false }).Count -eq 0
+
+  return $allValid
 }
 
-function DockerValidation($packageName, $packageVersion, $groupId, $DocValidationImageId, $workingdirectory) 
-{
-  $output = docker run -v "${workingDirectory}:/workdir/out" `
-    -e TARGET_PACKAGE=$packageName -e TARGET_VERSION=$packageVersion -e TARGET_GROUP_ID=$groupId -t $DocValidationImageId 2>&1 
-  # The docker exit codes: https://docs.docker.com/engine/reference/run/#exit-status
-  # If the docker failed because of docker itself instead of the application, 
-  # we should skip the validation and keep the packages. 
-  $artifactNamePrefix = "${groupId}:${packageName}:${packageVersion}"
-  if ($LASTEXITCODE -eq 125 -Or $LASTEXITCODE -eq 126 -Or $LASTEXITCODE -eq 127) 
-  { 
-    LogWarning "The `docker` command does not work with exit code $LASTEXITCODE. Fall back to mvn install $artifactNamePrefix directly."
-    $output | Write-Host
-    FallbackValidation -artifactNamePrefix "$artifactNamePrefix" -workingDirectory $workingdirectory
-  }
-  elseif ($LASTEXITCODE -ne 0) 
-  { 
-    LogWarning "Package $artifactNamePrefix ref docs validation failed."
-    $output | Write-Host
+function DockerValidation ($packageInfos, $DocValidationImageId, $workingDirectory) {
+  Write-Host "Validating $($packageInfos.Length) package(s) using $DocValidationImageId."
+
+  $containerWorkingDirectory = '/workdir/out'
+  $configurationFileName = 'configuration.json'
+
+  $hostConfigurationPath = Join-Path $workingDirectory $configurationFileName
+
+  # Cannot use Join-Path because the container and host path separators may differ
+  $containerConfigurationPath = "$containerWorkingDirectory/$configurationFileName"
+  
+  $configuration = [ordered]@{
+    "output_path" = "docs-ref-autogen";
+    "packages" = @($packageInfos | ForEach-Object { [ordered]@{ 
+        packageGroupId = $_.Group;
+        packageArtifactId = $_.Name;
+        packageVersion = $_.Version;
+        packageDownloadUrl = 'https://repo1.maven.org/maven2'
```

Comment:
This shouldn't be hard coding the Url, there's already a global variable defined at the top of the package for this, $packageDownloadUrl.

Reply:
That property isn't used, but is required by msdocs' script.   It ignores that url and uses the dev feed anyway.

Reply:
Sure, but just in case let's use the global instead of hard coding so we're at least consistently ignoring the same variable :)

Reply:
done